src/audio/codec_adapter/codec/generic.c

```diff
@@ -346,3 +346,106 @@ int module_free(struct processing_module *mod)
 
 	return ret;
 }
+
+/**
+ * \brief Set module configuration - Common method to assemble large configuration message
+ * \param[in] mod - struct processing_module pointer
+ * \param[in] config_id - Configuration ID
+ * \param[in] pos - position of the fragment in the large message
+ * \param[in] data_offset_size: size of the whole configuration if it is the first fragment or the
+ *				 only fragment. Otherwise, it is the offset of the fragment in the
+ *				 whole configuration.
+ * \param[in] fragment: configuration fragment buffer
+ * \param[in] fragment_size: size of @fragment
+ * \params[in] response: optional response buffer to fill
+ * \params[in] response_size: size of @response
+ *
+ * \return: 0 upon success or error upon failure
+ */
+int module_set_configuration(struct processing_module *mod,
+			     uint32_t config_id,
+			     enum module_cfg_fragment_position pos, size_t data_offset_size,
+			     const uint8_t *fragment, size_t fragment_size, uint8_t *response,
+			     size_t response_size)
+{
+	struct module_data *md = &mod->priv;
+	struct comp_dev *dev = mod->dev;
+	static size_t size;
+	size_t offset = 0;
+	uint8_t *dst;
+	int ret;
+
+	switch (pos) {
+	case MODULE_CFG_FRAGMENT_FIRST:
+	case MODULE_CFG_FRAGMENT_SINGLE:
+		/*
+		 * verify input params & allocate memory for the config blob when the first
+		 * fragment arrives
+		 */
+		size = data_offset_size;
+
+		/* Check that there is no previous request in progress */
+		if (md->runtime_params) {
+			comp_err(dev, "module_set_configuration(): error: busy with previous request");
+			return -EBUSY;
+		}
+
+		if (!size) {
+			comp_err(dev, "module_set_configuration(): error: no configuration size %d",
+				 size);
+			return -EINVAL;
+		}
+
+		if (size > MAX_BLOB_SIZE) {
+			comp_err(dev, "module_set_configuration(): error: blob size is too big cfg size %d, allowed %d",
+				 size, MAX_BLOB_SIZE);
+			return -EINVAL;
+		}
+
+		/* Allocate buffer for new params */
+		md->runtime_params = rballoc(0, SOF_MEM_CAPS_RAM, size);
+		if (!md->runtime_params) {
+			comp_err(dev, "module_set_configuration(): space allocation for new params failed");
+			return -ENOMEM;
+		}
+
+		memset(md->runtime_params, 0, size);
+		break;
+	default:
+		if (!md->runtime_params) {
+			comp_err(dev, "module_set_configuration(): error: no memory available for runtime params in consecutive load",
+				 ret);
+			return -EIO;
+		}
+
+		/* set offset for intermediate and last fragments */
+		offset = data_offset_size;
+		break;
+	}
+
+	dst = (uint8_t *)md->runtime_params + offset;
+
+	ret = memcpy_s(dst, size - offset, fragment, fragment_size);
```

Comment:
I don't understand how this compiles without warnings. `size` **is uninitialised** if you take the `default` path in `switch` above

Reply:
size is a static variable isnt it? Are they initialized to 0? besides it should be set to the total size with the first fragment in practice no?

Reply:
ouch, my bad. static variables are (fortunately) very rarely used in functions so I overlooked that. And it isn't good to use that here and no reason for that. Please, move it to a `struct`

Reply:
yes, i will remove it in a follow up PR